token/services/interop/pledge/wallet.go

```diff
@@ -0,0 +1,236 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package pledge
+
+import (
+	"encoding/json"
+
+	"github.com/hyperledger-labs/fabric-smart-client/platform/view"
+	view2 "github.com/hyperledger-labs/fabric-smart-client/platform/view/view"
+	"github.com/hyperledger-labs/fabric-token-sdk/token"
+	"github.com/hyperledger-labs/fabric-token-sdk/token/driver"
+	"github.com/hyperledger-labs/fabric-token-sdk/token/services/identity"
+	"github.com/hyperledger-labs/fabric-token-sdk/token/services/network"
+	token2 "github.com/hyperledger-labs/fabric-token-sdk/token/token"
+	"github.com/pkg/errors"
+)
+
+func MyOwnerWallet(sp view.ServiceProvider) (*token.OwnerWallet, error) {
+	w := token.GetManagementService(sp).WalletManager().OwnerWallet("")
+	if w == nil {
+		return nil, errors.Errorf("owner Wallet needs to be initialized")
+	}
+	return w, nil
+}
+
+func GetOwnerWallet(sp view.ServiceProvider, id string, opts ...token.ServiceOption) (*token.OwnerWallet, error) {
+	w := token.GetManagementService(sp, opts...).WalletManager().OwnerWallet(id)
+	if w == nil {
+		return nil, errors.Errorf("owner Wallet needs to be initialized")
+	}
+	return w, nil
+}
+
+func GetIssuerWallet(sp view.ServiceProvider, id string, opts ...token.ServiceOption) (*token.IssuerWallet, error) {
+	w := token.GetManagementService(sp, opts...).WalletManager().IssuerWallet(id)
+	if w == nil {
+		return nil, errors.Errorf("issuer Wallet needs to be initialized")
+	}
+	return w, nil
+}
+
+type QueryService interface {
+	// TODO: switch to UnspentTokensIteratorBy(id, typ string) (UnspentTokensIterator, error)
+	ListUnspentTokens() (*token2.UnspentTokens, error)
+}
+
+type IssuerWallet struct {
```

Comment:
The functionality is the same for `OwnerWallet`. I would create one type `wallet` and then
```
type IssuerWallet wallet
type OwnerWallet wallet
```